### ISSUE TITLE
Theme imports of a pnpm-linked package resolve from the symlink, not the real path, so the theme fails to build and no-unknown-classes silently degrades

### ISSUE THREAD

**Issue by leandromatos** [shadcn-ui/lint#15]
### What happens

When the theme imports a package installed by pnpm, that package is a symlink into `node_modules/.pnpm/`, and its own dependencies are installed beside the real file, not beside the link. `resolveStylesheet` returns the linked path, so the next `@import` inside that file walks up from the link, finds nothing, and the theme fails to build:

```text
[@shadcn/lint] The Tailwind theme at src/styles/globals.css could not be built (@import "kit-font/400.css" could not be resolved from /path/to/app/node_modules/linked-kit/src/styles); no-unknown-classes is using the grammar bundled with @shadcn/lint there until it can.
```

From then on `no-unknown-classes` judges every class against the bundled grammar: a `@utility` the project declares is reported as unknown, and a misspelled theme token is not. Same surface as #9, third cause; #9 is an `exports` pattern, #13 a partial without Tailwind, this one is where the file lives.

### Reproduction

A design-system package that ships its stylesheet and imports a dependency of its own from it, installed by pnpm (default isolated `node-linker`):

```jsonc
// node_modules/.pnpm/linked-kit@1.0.0/node_modules/linked-kit/package.json
{ "name": "linked-kit", "version": "1.0.0", "dependencies": { "kit-font": "1.0.0" }, "exports": { "./css": "./src/styles/kit.css" } }
```

```css
/* node_modules/.pnpm/linked-kit@1.0.0/node_modules/linked-kit/src/styles/kit.css */
@import "kit-font/400.css";
@utility kit-frame { outline: 1px dashed currentColor; }
```

`kit-font` is installed beside the real package, at `node_modules/.pnpm/linked-kit@1.0.0/node_modules/kit-font/`, and `node_modules/linked-kit` is a symlink to `.pnpm/linked-kit@1.0.0/node_modules/linked-kit`.

```css
/* src/app.css, the tailwind.css entry */
@import "tailwindcss";
@import "linked-kit/css";
```

```tsx
export function Page() { return <div className="kit-frame">Framed</div> }
```

**Expected:** `kit-font/400.css` resolves from `.pnpm/linked-kit@1.0.0/node_modules/`, the theme builds, `kit-frame` is accepted.

**Actual:** the stderr line above, exit code 0, `kit-frame` reported as unknown. Tailwind's own bundler resolves the same imports without trouble, so the build works and the linter disagrees with it.

### Cause

`loadStylesheet` in `src/tailwind/oracle.ts` uses the path `resolveStylesheet` returns as the base for the file's own imports:

```ts
const file = resolveStylesheet(base, id)
files.push(file)
return { base: path.dirname(file), content: fs.readFileSync(file, "utf-8") }
```

`resolveStylesheet` finds the package through the link and never follows it, so `base` is `node_modules/linked-kit/src/styles` and `packageDirectory` walks up from there.

### Fix

Resolve the imported file to its real path before using its directory as the base:

```ts
const resolved = resolveStylesheet(base, id)
if (!resolved) throw new Error(`@import "${id}" could not be resolved from ${base}`)
const file = fs.realpathSync.native(resolved)
```

The fix, a fixture in the pnpm shape and a regression test in `test/tailwind-oracle.test.ts` are on [`leandromatos/lint@fix/stylesheet-realpath`](https://github.com/leandromatos/lint/tree/fix/stylesheet-realpath), with a patch changeset; the full suite, typecheck and lint pass. I tried to open it as a pull request and GitHub refused (`leandromatos does not have the correct permissions to execute CreatePullRequest`), so it is here as a branch you can cherry-pick.
